### PR TITLE
fix(useEllipsisDetection): add debounce

### PR DESCRIPTION
**Pull request by GaroGabrielyan**

### CODE REVIEW

File: src/hooks/useWindowSize.js [hunk 1]
```diff
@@ -5,19 +5,26 @@ function useWindowSize() {
...
     const [height, setWindowHeight] = useState(innerHeight);
+    let timeoutId;
 
     const handleResize = () => {
         const { innerHeight, innerWidth } = w;
         setWindowWidth(innerWidth);
         setWindowHeight(innerHeight);
     };
 
+    const debounceResize = () => {
```
Comment (NarekArshakyan) [inline src/hooks/useWindowSize.js:16]: you repeat the same function 2 times, you can upgrade the current hook or create a new one

File: stories/atoms/Icon/Icon.stories.jsx [hunk 2]
```diff
@@ -103,57 +107,54 @@ Icon.args = {
     type: 'bc-icon-sound'
 };
 
+const IconCard = ({ name, copyContent }) => {
```
Comment (NarekArshakyan) [inline stories/atoms/Icon/Icon.stories.jsx:110]: This is a different component, you can write it in another file.

File: stories/atoms/Icon/Icon.stories.jsx [hunk 3]
```diff
@@ -1,32 +1,36 @@
...
+import {
+    BusyLoader,
+    Empty,
+    ExtendedInput,
+    Tooltip,
+    useEllipsisDetection,
+    Toaster,
+    Icon as IconComponent
+} from '../../../src';
 import './Icon.stories.scss';
```
Comment (NarekArshakyan) [inline stories/atoms/Icon/Icon.stories.jsx:13]: // Styles

File: stories/atoms/Icon/Icon.stories.jsx [hunk 4]
```diff
@@ -1,32 +1,36 @@
-import React, { useEffect, useState, createRef } from 'react';
-
-import IconComponent from 'src/lib/atoms/Icon/index';
-import ToasterComponent from 'src/lib/organisms/Toaster/index';
+import React, { useEffect, useState, useRef } from 'react';
+
+import { args, propCategory } from '../../assets/storybook.globals';
+import {
```
Comment (NarekArshakyan) [inline stories/atoms/Icon/Icon.stories.jsx:4]: // Components

File: stories/atoms/Icon/Icon.stories.jsx (hunk 2 above)
Comment (GaroGabrielyan) [reply stories/atoms/Icon/Icon.stories.jsx:110]: For now, it's best to keep the current file structure unchanged. We may reconsider this approach in the future if the need arises.

**Review by NarekArshakyan — DISMISSED** [review]

**Pull request merged**